urlac47192b66_ggrandes_bouncer_tgz-pJ8-org_javastack_bouncer_ClusterTestJ8/src/org/javastack/bouncer/Options.java

```diff
@@ -0,0 +1,348 @@
+	public static final String P_READ_TIMEOUT = "READ_TIMEOUT";
+	public static final String P_MUX_NAME = "MUX_NAME";
+	public static final String P_TUN_ID = "TUN_ID";
+	public static final String P_STICKY = "STICKY";
+	public static final String P_CLUSTER_ID = "CLUSTER_ID";
+	//
+	@SuppressWarnings("serial")
+	private final static Map<String, Integer> MAP_FLAGS = Collections
+			.unmodifiableMap(new HashMap<String, Integer>() {
+				{
+					put("LB=ORDER", LB_ORDER);
+					put("LB=RR", LB_RR);
+					put("LB=RAND", LB_RAND);
+					put("TUN=SSL", TUN_SSL);
+					put("TUN=ENDSSL", TUN_ENDSSL);
+					put("MUX=OUT", MUX_OUT);
+					put("MUX=IN", MUX_IN);
+					put("MUX=AES", MUX_AES);
+					put("MUX=SSL", MUX_SSL);
+					put("PROXY=SEND", PROXY_SEND);
+					put("CLUSTER=AES", CLUSTER_AES);
+					put("CLUSTER=SSL", CLUSTER_SSL);
+				}
+			});
+	//
+	StickyConfig stickyConfig = null;
+	int flags;
+	@SuppressWarnings("serial")
+	final Map<String, String> strParams = Collections.synchronizedMap(new HashMap<String, String>() {
+		{
+			put(P_MUX_NAME, S_NULL); 	// MUX_NAME=<muxName>
+			put(P_AES, S_NULL); 		// AES=<key>
+			put(P_AES_ALG, S_NULL); 	// AESALG=<cipherAlgorithm>
+			// SSL=server.crt:server.key:client.crt (MUX-IN|CLUSTER-IN)
+			// SSL=client.crt:client.key:server.crt (MUX-OUT|CLUSTER-OUT)
+			put(P_SSL, S_NULL);
+			// ENDSSL=server.crt:server.key[:client.crt]
+			put(P_ENDSSL, S_NULL);
+			put(P_STICKY, S_NULL);		// STICKY=MEM:bitmask:elements:ttl
+		}
+	});
+	@SuppressWarnings("serial")
+	final Map<String, Integer> intParams = Collections.synchronizedMap(new HashMap<String, Integer>() {
+		{
+			put(P_CONNECT_TIMEOUT, I_NULL); // CONNECT_TIMEOUT=millis
+			put(P_READ_TIMEOUT, I_NULL); 	// READ_TIMEOUT=millis
+			put(P_TUN_ID, I_NULL); 			// TUN_ID=<idEndPoint>
+			put(P_AES_BITS, I_NULL); 		// AESBITS=<keyLengthInBits>
+		}
+	});
+
+	@SuppressWarnings("serial")
+	final Map<String, Long> longParams = Collections.synchronizedMap(new HashMap<String, Long>() {
+		{
+			put(P_CLUSTER_ID, L_NULL); 		// CLUSTER_ID=<idCluster>
+		}
+	});
+
+	public Options(final String strOpts) {
+		this.flags = parseOptions(strOpts);
+	}
+
+	// Clone Constructor
+	public Options(final Options old) {
+		this.flags = old.flags;
+		for (Entry<String, String> e : old.strParams.entrySet()) {
+			strParams.put(e.getKey(), e.getValue());
+		}
+		for (Entry<String, Integer> e : old.intParams.entrySet()) {
+			intParams.put(e.getKey(), e.getValue());
+		}
+		for (Entry<String, Long> e : old.longParams.entrySet()) {
+			longParams.put(e.getKey(), e.getValue());
+		}
+	}
+
+	public int getFlags(final int filterBits) {
+		return (flags & filterBits);
+	}
+
+	public Options setFlags(final int bits) {
+		flags |= bits;
+		return this;
+	}
+
+	public Options unsetFlags(final int bits) {
+		flags &= ~bits;
+		return this;
+	}
+
+	public String getString(final String name, final String def) {
+		final String value = strParams.get(name);
+		if (value == S_NULL) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.